src/erlcloud_aws.erl

```diff
@@ -266,21 +266,26 @@ format_timestamp({{Yr, Mo, Da}, {H, M, S}}) ->
 
 default_config() ->
     case get(aws_config) of
-        undefined ->
-            AccessKeyId = case os:getenv("AWS_ACCESS_KEY_ID") of
-                              false -> "";
-                              AKI -> AKI
-                          end,
-            SecretAccessKey = case os:getenv("AWS_SECRET_ACCESS_KEY") of
-                                  false -> "";
-                                  SAC -> SAC
-                              end,
+        undefined -> default_config_env();
+        Config -> Config
+    end.
+
+default_config_env() ->
+    case {os:getenv("AWS_ACCESS_KEY_ID"), os:getenv("AWS_SECRET_ACCESS_KEY")} of
+        {false, false} ->
+            default_config_profile();
+        {AccessKeyId, SecretAccessKey} ->
```

Comment:
similar to @kkuzmin comment this now can be `{"blah", false}` or `{false, "blah"}` which will break things and even not as it was before.

I would agree that this can be as simple as:
```erlang
case {os:getenv("AWS_ACCESS_KEY_ID"), os:getenv("AWS_SECRET_ACCESS_KEY")} of
    {KeyId, Secret} when is_list(KeyId) andalso is_list(Secret) -> 
        #aws_config{access_key_id = AccessKeyId,
                   secret_access_key = SecretAccessKey};
   _ -> default_config_profile()
end
```